internal/tiltfile/docker_compose.go

```diff
@@ -93,11 +93,12 @@ func (s *tiltfileState) dcResource(thread *starlark.Thread, fn *starlark.Builtin
 		return nil, fmt.Errorf("dc_resource: `name` must not be empty")
 	}
 
-	var imageRefAsStr string
+	var imageRefAsStr *string
```

Comment:
what's the point of distinguishing `nil` from the empty string? seems like you'd want them to behave the same.